caas/kubernetes/provider/bootstrap.go

```diff
@@ -64,15 +69,36 @@ type controllerStacker interface {
 	Deploy() error
 }
 
-func newcontrollerStack(stackName string, storageClass string, broker *kubernetesClient, pcfg *podcfg.ControllerPodConfig) (controllerStacker, error) {
+// setControllerNamespace sets controller's namespace to name.
+// This is only used for bootstrap - set namespace from `controller` to `controller-name`.
+func setControllerNamespace(name string, broker *kubernetesClient) error {
+	_, err := broker.GetNamespace(name)
+	if errors.IsNotFound(err) {
+		// all good.
+		broker.namespace = name
+		return nil
+	}
+	if err == nil {
+		// this should never happen because we avoid it in broker.PrepareForBootstrap before reaching here.
+		return errors.NotValidf("existing namespace %q found", broker.namespace)
+	}
+	return errors.Trace(err)
+}
+
+func newcontrollerStack(
+	stackName string,
+	storageClass string,
+	broker *kubernetesClient,
+	pcfg *podcfg.ControllerPodConfig,
+) (controllerStacker, error) {
 	// TODO(caas): parse from constrains?
```

Comment:
there's a root-disk constraint we should use
can you add a trello card

Reply:
yeah, created https://trello.com/c/PLfDt6t3